.github/workflows/build_docker_images.yml

```diff
@@ -40,12 +40,13 @@ jobs:
           password: ${{ secrets.DOCKER_TOKEN }}
 
       - name: "Checkout"
-        uses: actions/checkout@v3
+        uses: actions/checkout@v4
 
       - name: "Paths filter"
-        uses: dorny/paths-filter@v2
+        uses: dorny/paths-filter@v3
         id: changes
         with:
+          base: ${{ github.ref }}
```

Comment:
Ah, the magic. Thank you and @chiaberry for y'all's insight in this. I was struggling to understand all the moving parts of this git operation and what was going on, to say the least. 🙏